src/aff_expr.jl

```diff
@@ -809,3 +809,74 @@ moi_function(a::Vector{<:GenericAffExpr}) = MOI.VectorAffineFunction(a)
 function moi_function_type(::Type{<:Vector{<:GenericAffExpr{T}}}) where {T}
     return MOI.VectorAffineFunction{T}
 end
+
+"""
+    _eval_as_variable(f::F, x::GenericAffExpr, args...) where {F}
+
+In many cases, `@variable` can return a `GenericAffExpr` instead of a
+`GenericVariableRef`. This is particularly the case for complex-valued
+expressions. To make common operatons like `lower_bound(x)` work, we should
+forward the method if and only if `x` is convertable to a `GenericVariableRef`.
+"""
+function _eval_as_variable(f::F, x::GenericAffExpr, args...) where {F}
+    if length(x.terms) != 1
+        error(
+            "Cannot call $f with $x because it is not a real-valued affine " *
```

Comment:
Not sure we need the `real-value`. the `+1` below is enough. Otherwise, the user might think the type needs to be `Real` while `one(ComplexF64)` actually works as coefficient. It's not about the type, it's just about being equal to `1`.

Reply:
Ah we remove the complex below, never mind